srml/elections-phragmen/src/lib.rs

```diff
@@ -0,0 +1,952 @@
+		pub VotesOf get(votes_of): linked_map T::AccountId => (Vec<T::AccountId>, u32);
+		/// Locked stake of a voter.
+		pub StakeOf get(stake_of): map T::AccountId => BalanceOf<T>;
+
+		/// The present candidate list.
+		pub Candidates get(candidates): Vec<T::AccountId>;
+		/// Current number of active candidates.
+		pub CandidateCount get(candidate_count): u32;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		fn deposit_event<T>() = default;
+
+		const CandidacyBond: BalanceOf<T> = T::CandidacyBond::get();
+		const VotingBond: BalanceOf<T> = T::VotingBond::get();
+
+		/// Vote for a set of candidates for the upcoming round of election.
+		///
+		/// The `votes` should:
+		///   - not be empty.
+		///   - not be less than the number of candidates.
+		///
+		/// `who` cannot be an old voter who has not yet called [`remove_voter`].
+		///
+		/// Upon voting, the entire balance of `who` is locked and a bond amount is reserved.
+		#[weight = SimpleDispatchInfo::FixedNormal(2_500_000)]
+		fn vote(origin, votes: Vec<T::AccountId>) -> Result {
+			let who = ensure_signed(origin)?;
+
+			// TODO: use decode_len #3071.
+			let candidates = Self::candidates();
+			ensure!(!candidates.is_empty(), "number of candidates cannot be zero");
+			ensure!(candidates.len() >= votes.len(), "cannot vote more than candidates");
+
+			ensure!(!votes.is_empty(), "must vote for at least one candidate.");
+			ensure!(!Self::is_old_voter(&who), "must first remove self as voter and re-submit");
+
+			if !Self::is_current_voter(&who) {
+				// Amount to be locked up.
+				let locked_balance = T::Currency::total_balance(&who);
+
+				// lock and reserve
+				T::Currency::reserve(&who, T::VotingBond::get())
+					.map_err(|_| "voter can not pay voting bond")?;
+				T::Currency::set_lock(
+					MODULE_ID,
+					&who,
+					locked_balance,
+					T::BlockNumber::max_value(),
+					WithdrawReasons::except(WithdrawReason::TransactionPayment),
+				);
+				<StakeOf<T>>::insert(&who, locked_balance);
+			}
+
+			let now = Self::election_rounds();
+			<VotesOf<T>>::insert(&who, (votes, now));
+			Ok(())
+		}
+
+		/// Remove `origin` as a voter. One can use this to
+		///   1. _undo_ their votes before the election has started.
+		///   2. Remove their lock and unreserve the bond after an election round.
+		#[weight = SimpleDispatchInfo::FixedNormal(1_250_000)]
+		fn remove_voter(origin) {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				Self::is_current_voter(&who) || <StakeOf<T>>::exists(&who),
+				"should be a current/old voter"
+			);
+
+			// remove storage.
+			<VotesOf<T>>::remove(&who);
+			<StakeOf<T>>::remove(&who);
+
+			// unreserve the bond and remove.
+			T::Currency::unreserve(&who, T::VotingBond::get());
+			T::Currency::remove_lock(MODULE_ID, &who);
+		}
+
+		/// Submit oneself for candidacy.
+		///
+		/// A candidate will either:
+		///   - Lose at the end of the term and gets their bond unreserved
```

Comment:
```suggestion
		///   - Lose at the end of the term and gets their bond unreserved.
```

And I suppose maybe this is too lenient and we should burn/keep the bond for loser candidates? we do the same for the normal approval based election (though it has an issue #3363,but at least the intention is to not return the bond)

Reply:
This is already updated and we handle the loser bonds via an imbalance.